packages/slope/src/adapter.ts

```diff
@@ -0,0 +1,191 @@
+import {
+    BaseSignerWalletAdapter,
+    pollUntilReady,
+    WalletAccountError,
+    WalletNotConnectedError,
+    WalletNotFoundError,
+    WalletNotInstalledError,
+    WalletPublicKeyError,
+    WalletSignTransactionError,
+} from '@solana/wallet-adapter-base';
+import { PublicKey, Transaction } from '@solana/web3.js';
+import bs58 from 'bs58';
+
+interface SlopeWallet {
+    connect(): Promise<{
+        msg: string;
+        data: {
+            publicKey: string;
+        };
```

Comment:
Will `data` be defined even if `msg != 'ok'`?

Reply:
When `msg != 'ok'`, `data = {}`

Reply:
Got it. Thanks!